images/controller-engine-v2.svg

```diff
@@ -0,0 +1,621 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```

Comment:
I'm unable to read the graph due to rendering issue - how about replacing this with a Mermaid chart in the proposal directly?

(Please note I may have failed to transcribe it correctly...)

```mermaid
flowchart LR
    %% Define front-end web components
    subgraph io["IO"]
        co["Control Object"]
        midiIO["Midi IO"]
        hidIO["HID IO"]
        mixxxLib["Mixxx Library"]
        mixxxUi["Mixxx UI"]
    end
    style io fill:transparent,stroke:green,color:#fff

    subgraph layers["Intermediary Layers"]
        co---midiDispatcher["MIDI Dispatcher"]
        midiIO---midiDispatcher
        mixxxUi---uiShift["UI Shit Capability"]
    end
    style layers fill:transparent,stroke:yellow,color:#fff

    subgraph proxies["API Proxies"]
        co---coJSProxy["COJSProxy"]
        co---coQMLProxy["COQmlProxy"]
        midiIO---midiJSProxy["MIDI IO JSProxy"]
        midiIO---midiQMLProxy["MIDI IO QmlProxy"]
        midiDispatcher---midiDispatcherJSProxy["MIDI Dispatcher JSProxy"]
        mixxxLib---libQmlProxy["Library QmlProxy"]
    end
    style proxies fill:transparent,stroke:cyan,color:#fff

    uiShift --- proxies

    subgraph engine["Execution Engine"]
        coJSProxy---jsEngine["JSExecution Engine"]
        midiJSProxy---jsEngine
        midiDispatcherJSProxy---jsEngine
        coJSProxy---jsLegacyEngine["LegacyJS Execution Engine"]
        midiJSProxy---jsLegacyEngine
        coQMLProxy---qmlEngine["QmlExecutionEngine"]
        midiQMLProxy---qmlEngine
        libQmlProxy---qmlEngine
    end
    proxies --- engine
    style engine fill:transparent,stroke:red,color:#fff
```

Reply:
Thank you, that looks good (apart from the typo in "shift" :sweat_smile:). I didn't have a renderer handy for previewing mermaid diagrams (I also didn't want to learn the syntax just to get this PR out right now) so I didn't chose that initially. I will replace the svg with your diagram in the next few commits.